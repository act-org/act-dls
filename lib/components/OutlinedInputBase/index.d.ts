/**
 * @prettier
 */
import * as React from 'react';
import { OutlinedInputProps } from '@material-ui/core';
export declare type Props = OutlinedInputProps;
declare const OutlinedInputBase: React.FC<Props>;
export default OutlinedInputBase;
//# sourceMappingURL=index.d.ts.map