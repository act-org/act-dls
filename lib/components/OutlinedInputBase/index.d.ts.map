{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/OutlinedInputBase/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAC/B,OAAO,EAAiB,kBAAkB,EAAE,MAAM,mBAAmB,CAAC;AAEtE,oBAAY,KAAK,GAAG,kBAAkB,CAAC;AAEvC,QAAA,MAAM,iBAAiB,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAEmB,CAAC;AAE3D,eAAe,iBAAiB,CAAC"}