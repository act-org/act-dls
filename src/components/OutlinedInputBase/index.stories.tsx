/**
 * @prettier
 */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';

import OutlinedInputBase from '.';

export const Base = (): React.ReactElement<any> => {
  const [value, setValue] = React.useState('OutlinedInput');

  return (
    <OutlinedInputBase
      autoFocus={boolean('Auto Focus?', false)}
      color={select(
        'Color',
        {
          Primary: 'primary',
          Secondary: 'secondary',
        },
        'primary',
      )}
      disabled={boolean('Disabled?', false)}
      multiline={boolean('Multiple Lines?', false)}
      onChange={(e: any): void => {
        action('onChange()');

        setValue(e.target.value);
      }}
      placeholder={text('Placeholder', 'Placeholder')}
      value={value}
    />
  );
};

export default {
  component: OutlinedInputBase,
  parameters: {
    info: {
      text: 'This is the base variant of the OutlinedInput component.',
    },
  },
  title: 'Atoms/OutlinedInput',
};
