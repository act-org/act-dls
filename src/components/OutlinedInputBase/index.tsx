/**
 * @prettier
 */

import * as React from 'react';
import { OutlinedInput, OutlinedInputProps } from '@material-ui/core';

export type Props = OutlinedInputProps;

const OutlinedInputBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <OutlinedInput {...props} />;

export default OutlinedInputBase;
