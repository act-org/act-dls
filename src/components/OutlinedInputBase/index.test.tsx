/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import OutlinedInputBase from '.';

describe('OutlinedInputBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <OutlinedInputBase onChange={noop} value="Value" />,
    );

    expect(container).toMatchSnapshot();
  });
});
