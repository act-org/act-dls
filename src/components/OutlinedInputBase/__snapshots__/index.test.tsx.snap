// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlinedInputBase matches the snapshot 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root"
  >
    <input
      class="MuiInputBase-input MuiOutlinedInput-input"
      type="text"
      value="Value"
    />
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      style="padding-left: 8px;"
    >
      <legend
        class="PrivateNotchedOutline-legend-2"
        style="width: 0.01px;"
      >
        <span>
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
